### PR TITLE
Differentiate descriptions of Address and Endpoint

### PR DESCRIPTION
**Pull request by MarianMacik**
As discussed with @davidesalerno and @ricardozanini these two fields need better differentiation for users to understand.

<details>
<summary>
How to backport a pull request to a different branch?
</summary>

In order to automatically create a **backporting pull request** please add one or more labels having the following format `backport-<branch-name>`, where `<branch-name>` is the name of the branch where the pull request must be backported to (e.g., `backport-7.67.x` to backport the original PR to the `7.67.x` branch).

> **NOTE**: **backporting** is an action aiming to move a change (usually a commit) from a branch (usually the main one) to another one, which is generally referring to a still maintained release branch. Keeping it simple: it is about to move a specific change or a set of them from one branch to another.

Once the original pull request is successfully merged, the automated action will create one backporting pull request per each label (with the previous format) that has been added.

If something goes wrong, the author will be notified and at this point a manual backporting is needed.

> **NOTE**: this automated backporting is triggered whenever a pull request on `main` branch is labeled or closed, but both conditions must be satisfied to get the new PR created.
</details>

### CODE REVIEW

File: api/v1alpha08/sonataflow_types.go [hunk 1]
```diff
@@ -120,12 +120,14 @@ type SonataFlowSpec struct {
 // SonataFlowStatus defines the observed state of SonataFlow
 type SonataFlowStatus struct {
 	api.Status `json:",inline"`
+	// Used as a part of Addressable interface (status.address.url) for knative
```
Comment (spolti) [inline api/v1alpha08/sonataflow_types.go:123]:
```suggestion
	// Address is used as a part of Addressable interface (status.address.url) for knative
```

File: api/v1alpha08/sonataflow_types.go [hunk 2]
```diff
@@ -120,12 +120,14 @@ type SonataFlowSpec struct {
...
 	// +optional
 	Address duckv1.Addressable `json:"address,omitempty"`
 	// keeps track of how many failure recovers a given workflow had so far
 	RecoverFailureAttempts int         `json:"recoverFailureAttempts,omitempty"`
 	LastTimeRecoverAttempt metav1.Time `json:"lastTimeRecoverAttempt,omitempty"`
-	Endpoint               *apis.URL   `json:"endpoint,omitempty"`
+	// Externally accessible URL of the workflow
```
Comment (spolti) [inline api/v1alpha08/sonataflow_types.go:129]:
```suggestion
	// Endpoint is an externally accessible URL of the workflow
```

**Review by ricardozanini — APPROVED** [review]
+1 after merging @spolti's comment

**Comment by MarianMacik** [comment]
Done. One question though. Why do we have to repeat it? The description is then unnecessarily repeating the field in the yaml. Other fields like `RecoverFailureAttempts` don't repeat it.

**Comment by spolti** [comment]
> Done. One question though. Why do we have to repeat it? The description is then unnecessarily repeating the field in the yaml. Other fields like `RecoverFailureAttempts` don't repeat it.

It is a best practice https://github.com/kiegroup/kogito-serverless-operator/pull/179#issuecomment-1615122364
About the other fields, we could aligning them slowly :)

**Review by spolti — APPROVED** [review]

**Review by davidesalerno — APPROVED** [review]

**Pull request merged**